integration/spark/src/main/scala/org/apache/carbondata/api/CarbonStore.scala

```diff
@@ -96,20 +100,31 @@ object CarbonStore {
    * Read stage files and return input files
    */
   def readStageInput(
+      tableStagePath: String,
       stageFiles: Seq[CarbonFile],
       status: StageInput.StageStatus): Seq[StageInput] = {
     val gson = new Gson()
     val output = Collections.synchronizedList(new util.ArrayList[StageInput]())
     stageFiles.map { stage =>
-      val filePath = stage.getAbsolutePath
-      val stream = FileFactory.getDataInputStream(filePath)
+      val filePath = tableStagePath + CarbonCommonConstants.FILE_SEPARATOR + stage.getName
+      var stream: DataInputStream = null
       try {
-        val stageInput = gson.fromJson(new InputStreamReader(stream), classOf[StageInput])
-        stageInput.setCreateTime(stage.getLastModifiedTime)
-        stageInput.setStatus(status)
-        output.add(stageInput)
+        stream = FileFactory.getDataInputStream(filePath)
+        var retry = READ_FILE_RETRY_TIMES
+        breakable { while (retry > 0) { try {
+          val stageInput = gson.fromJson(new InputStreamReader(stream), classOf[StageInput])
+          stageInput.setCreateTime(stage.getLastModifiedTime)
+          stageInput.setStatus(status)
+          output.add(stageInput)
+        } catch {
+          case _ : FileNotFoundException => breakable()
```

Comment:
should add log if file is not found?

Reply:
I have modified code according to your suggestion